script.js

```diff
@@ -1,4 +1,230 @@
+//Cloned from day9 more dice more players
+
+//GLOBAL VARIABLES
+//Game states
+var START = "START";
+var SET_PLAYERS = "SET_PLAYERS";
+var SET_ROLLS = "SET_ROLLS";
+var DICE_ROLL = "DICE_ROLL";
+var SCOREBOARD = "SCOREBOARD";
+
+//Selected number of players and dice to play with
+var playersList = [];
+var chosenDiceRolls = 0;
+
+//Set initial state
+var gameState = START;
+var playerInPlay;
+var roundWinner;
+var overallWinner;
+
+//HELPER FUNCTIONS
+//create an array of objects - # of objects = # of players
+var createPlayersList = function (input) {
+  var currentPlayer = 1;
+
+  while (currentPlayer <= input) {
+    var player = {
+      name: `Player ${currentPlayer}`,
+      hand: [],
+      tally: 0,
+      score: 0,
+    }
+    playersList.push(player);
+    currentPlayer +=1;
+  }
+  return playersList;
+};
+
+
+//roll dice (1 to 6) for as many times as chosenDiceRolls indicates
+var rollDiceForPlayer = function () {
+  var counter = 0;
+
+  while (counter < chosenDiceRolls) {
+    playerInPlay.hand.push(Math.floor(Math.random() * 6) + 1);
+    counter += 1;
+  }
+  return playerInPlay.hand;
+};
+
+//sort largest to smallest number
+var sortBigToSmall = function (input) {
+
+  for (i = 0; i < input.length; i += 1) {
+    for (j = 0; j < (input.length - i - 1); j ++) {
+      if (input[j] < input[j+1]) {
+        var temp = input[j];
+        input[j] = input[j+1];
+        input[j+1] = temp;
+      } 
+    }
+      console.log(input)
+      console.log(j)
+      console.log(i)
+  }
+  return null;
+}
+
+//Converts array of numbers to a number
+var convertArrayToNumber = function (input) {
+  var runDown = chosenDiceRolls - 1;
+  var counter = 0;
+
+  while (runDown >= 0) {
+    var number = input[counter] * Math.pow(10, runDown);
+    console.log(number);
+    playerInPlay.tally += number;
+    console.log(playerInPlay.tally);
+    runDown -= 1;
+    counter += 1;
+  }
+  return playerInPlay.tally;
+};
+
+//Compares the score 
+var tallyComparison = function () {
+  var i = 0;
+  roundWinner = playersList[i];
+
+  while (i < playersList.length - 1) {
+    if (roundWinner.tally > playersList[i+1].tally) {
+      roundWinner = roundWinner;
+    } else {
+      roundWinner = playersList[i+1];
+    }
+    i +=1;
+  }
+
+  roundWinner.score +=1;
+  console.log (playersList.score);
+  return roundWinner;
```

Comment:
There are more advanced ways of finding the maximum of the scores, and finding the index of that particular winner, with the built-in javascript stuff like `Math.max` and `Array.indexOf`. It could make your code more compact! But this is a perfectly adequate way of finding who has the maximum tally!